src/dune_engine/dep.mli

```diff
@@ -26,20 +26,67 @@ val compare : t -> t -> Ordering.t
     necessitates a forward declaration to use in this module. *)
 val eval_pred : (File_selector.t -> Path.Set.t) Fdecl.t
 
-val peek_alias_expansion : (Alias.t -> Path.Set.t) Fdecl.t
+module Map : sig
+  type dep := t
 
-val eval_alias_expansion : (Alias.t -> Path.Set.t Memo.Build.t) Fdecl.t
+  include Map.S with type key := t
 
-module Trace : sig
+  val sandbox_config : _ t -> Sandbox_config.t
+
+  val has_universe : _ t -> bool
+
+  val parallel_map :
+    'a t -> f:(dep -> 'a -> 'b Memo.Build.t) -> 'b t Memo.Build.t
+end
+
+module Fact : sig
```

Comment:
It would be great to have a high-level overview of what facts are and how they are used in the computation of build traces. This seems to be a good place for such an overview but perhaps there is a better one.

Reply:
Agreed. Let me do that now while I'm zoomed in on this part of the code.

Reply:
I added something, WDYT?